websites/D/Duolingo/presence.ts

```diff
@@ -1,92 +1,232 @@
-lang.set("sv", "Swedish");
-lang.set("el", "Greek");
-lang.set("hi", "Hindi");
-lang.set("hv", "high valyrian");
-lang.set("ga", "Irish");
-lang.set("pl", "Polish");
-lang.set("he", "Hebrew");
-lang.set("nb", "Norwegian");
-lang.set("vi", "Vietnamese");
-lang.set("ar", "Arabic");
-lang.set("hw", "Hawaiian");
-lang.set("da", "Danish");
-lang.set("kl", "Klingon");
-lang.set("ro", "Romanian");
-lang.set("cs", "Czech");
-lang.set("sw", "Swahili");
-lang.set("cy", "Walsh");
-lang.set("id", "Indonesian");
-lang.set("hu", "Hungarian");
-lang.set("uk", "Ukrainian");
-lang.set("eo", "Esperanto");
-lang.set("nv", "Navajo");
-lang.set("en", "English");
+  language = { imageKey: "icon", name: "nothing (yet!)" };
 
-presence.on("UpdateData", async () => {
-  const path1 = document.location.pathname;
+function updateLanguage() {
+  const state = JSON.parse(window.localStorage.getItem("duo.state")),
+    courseId = state?.user?.currentCourseId;
+  if (!courseId) return;
+
+  const course = state.courses?.[courseId];
+  if (!course) return;
 
-  if (!path1.split("/")[2]) {
-    if (document.location.pathname.startsWith("/learn")) {
-      pageData.details = "Choosing level to learn..";
-      presence.setActivity(pageData);
-    } else if (document.location.pathname.startsWith("/shop")) {
-      pageData.details = "Browsing shop..";
-      presence.setActivity(pageData);
-    } else if (document.location.pathname.includes("/dictionary")) {
-      pageData.details = "Using dictionary..";
-      pageData.state = `Language: ${document.location.pathname.split("/")[2]}`;
-      presence.setActivity(pageData);
-    } else if (document.location.pathname.includes("/profile")) {
-      pageData.details = "Browsing profile..";
-      pageData.state = `Browsing: ${document.location.pathname.split("/")[2]}`;
-      presence.setActivity(pageData);
-    } else if (document.location.pathname.includes("/words")) {
-      presenceData.details = "Checking words...";
-      presenceData.largeImageKey = "logo";
-      presence.setActivity(presenceData);
-    } else if (
-      document.location.pathname === "/" ||
-      !document.location.pathname
-    )
-      presence.setActivity(pageData);
-  } else {
-    if (
-      path1.length > 1 &&
-      path1.split("/")[2] !== null &&
-      path1.split("/")[2].length === 2
-    ) {
-      let language: string;
-      for (const value of lang.keys()) {
-        if (path1.split("/")[2] === value) {
-          language = lang.get(value);
-          break;
-        }
-      }
-      presenceData.details = "Taking a lesson";
-      presenceData.state = `Language: ${language}`;
-      presenceData.largeImageKey = "logo";
-      presence.setActivity(presenceData);
+  const languageId = course.learningLanguage;
+  if (FLAG_ICONS.includes(languageId)) language.imageKey = `flag_${languageId}`;
+  else if (languageId) language.imageKey = "flag_unknown";
+  else language.imageKey = "icon";
+  language.name = course.title ?? "nothing (yet!)";
```

Comment:
Use a more apporopriate string instead of `nothing (yet!)`

Reply:
Do you have any suggestions? I personally don't see a problem with `nothing (yet!)`, it indicates that the user hasn't chosen a language yet and it fits grammatically:

![image](https://user-images.githubusercontent.com/64972126/142289803-c69959a1-5a93-4a59-8c53-ba47e20845e5.png)

Reply:
Set it to null